### DISCUSSION TITLE
Astro.js Project LSP Auto Complete Not Working

### DISCUSSION THREAD

**Discussion by JonasAllenCodes** [LazyVim/LazyVim#4980] [Extra Requests]
Hey I am new to NeoVim and been getting around well and hacked together what I need and everything seems to work well for all different type of projects. I have the Astro Language Extra installed. When it comes to Astro.js projects my LSP hints and Autocomplete does not seem to work. Also when I open an .astro file I have to toggle on Treesitter Highlight.

I tried :LspInfo and the below was printed. There seems to be an error with "root directory: Not found." Is this my issue? How do I fix this issue? 

Thank you in advance for any help 

==============================================================================
lspconfig: require("lspconfig.health").check()

LSP configs active in this session (globally) ~
- Configured servers: tailwindcss, bashls, dockerls, astro, yamlls, jsonls, taplo, vtsls, pyright, svelte, lua_ls, marksman, phpactor, eslint, ruff, docker_compose_language_service
- OK Deprecated servers: (none)

LSP configs active in this buffer (bufnr: 5) ~
- Language client log: ~/.local/state/nvim/lsp.log
- Detected filetype: `astro`
- 0 client(s) attached to this buffer
- Other clients that match the "astro" filetype:
- Config: tailwindcss
  filetypes:         aspnetcorerazor, astro, astro-markdown, blade, clojure, django-html, htmldjango, edge, eelixir, elixir, ejs, erb, eruby, gohtml, gohtmltmpl, haml, handlebars, hbs, html, htmlangular, html-eex, heex, jade, leaf, liquid, mdx, mustache, njk, nunjucks, php, razor, slim, twig, css, less, postcss, sass, scss, stylus, sugarss, javascript, javascriptreact, reason, rescript, typescript, typescriptreact, vue, svelte, templ
  cmd:               ~/.local/share/nvim/mason/bin/tailwindcss-language-server --stdio
  version:           `?` (Failed to get version) Tried:
  `/home/jonas/.local/share/nvim/mason/bin/tailwindcss-language-server --version`
  `/home/jonas/.local/share/nvim/mason/bin/tailwindcss-language-server -version`
  `/home/jonas/.local/share/nvim/mason/bin/tailwindcss-language-server version`
  `/home/jonas/.local/share/nvim/mason/bin/tailwindcss-language-server --help`
  
  executable:        true
  autostart:         true
  root directory:    ~/Documents/jonasallen_codes/website/
- Config: astro
  filetypes:         astro
  cmd:               ~/.local/share/nvim/mason/bin/astro-ls --stdio
  version:           `2.14.1`
  executable:        true
  autostart:         true
  root directory:    ~/Documents/jonasallen_codes/website/
- Config: eslint
  filetypes:         javascript, javascriptreact, javascript.jsx, typescript, typescriptreact, typescript.tsx, vue, svelte, astro
  cmd:               ~/.local/share/nvim/mason/bin/vscode-eslint-language-server --stdio
  version:           `?` (Failed to get version) Tried:
  `/home/jonas/.local/share/nvim/mason/bin/vscode-eslint-language-server --version`
  `/home/jonas/.local/share/nvim/mason/bin/vscode-eslint-language-server -version`
  `/home/jonas/.local/share/nvim/mason/bin/vscode-eslint-language-server version`
  `/home/jonas/.local/share/nvim/mason/bin/vscode-eslint-language-server --help`
  
  executable:        true
  autostart:         true
  root directory:    Not found.
  custom handlers:   eslint/probeFailed, eslint/confirmESLintExecution, eslint/openDoc, eslint/noLibrary

Docs for active configs: ~
- tailwindcss docs: >markdown
  
  https://github.com/tailwindlabs/tailwindcss-intellisense
  
  Tailwind CSS Language Server can be installed via npm:
  ```sh
  npm install -g @tailwindcss/language-server
  ```
  
  
- astro docs: >markdown
  
  https://github.com/withastro/language-tools/tree/main/packages/language-server
  
  `astro-ls` can be installed via `npm`:
  ```sh
  npm install -g @astrojs/language-server
  ```
  
  
- eslint docs: >markdown
  
  https://github.com/hrsh7th/vscode-langservers-extracted
  
  `vscode-eslint-language-server` is a linting engine for JavaScript / Typescript.
  It can be installed via `npm`:
  
  ```sh
  npm i -g vscode-langservers-extracted
  ```
  
  `vscode-eslint-language-server` provides an `EslintFixAll` command that can be used to format a document on save:
  ```lua
  lspconfig.eslint.setup({
    --- ...
    on_attach = function(client, bufnr)
      vim.api.nvim_create_autocmd("BufWritePre", {
        buffer = bufnr,
        command = "EslintFixAll",
      })
    end,
  })
  ```
  
  See [vscode-eslint](https://github.com/microsoft/vscode-eslint/blob/55871979d7af184bf09af491b6ea35ebd56822cf/server/src/eslintServer.ts#L216-L229) for configuration options.
  
  Messages handled in lspconfig: `eslint/openDoc`, `eslint/confirmESLintExecution`, `eslint/probeFailed`, `eslint/noLibrary`
  
  Additional messages you can handle: `eslint/noConfig`

**Comment by JonasAllenCodes** [comment]
Do I need the vscode extra? where do I put the config for the EslintFixAll? Why is it mentioning vscode when I am using nvim?

If someone can just point me in the right direction? I have done all the above steps but the config it says to add. When I try adding it into the config or plugins folder I get "undifined global `lspconfig'.", from the lsp warning along with "Unused local `client`. 

Do I need to specify my project root directory somewhere in my project?

Once again thank you for any help in this. I have looked at other discussions they say the issues are fixed with PRs but I am still having issues. I swear I had everything working when I was working on it about 2 months ago. Maybe it has to do with some updates? I know there were some major updates since then that changed the key bindings for comments. I used to be able to use "ctrl + /" in the past but now I have to use the mini.comment keybindings of "g then c" to make my selection a comment.

**Comment by zvrkSam** [comment]
For me astro works out of the box with extras, the only thing I add for every astro project is `prettier-plugin-astro` as dev dependency.
Did you try this approach ?

**Comment by JonasAllenCodes** [comment]
Okay i figured it out! I was missing the following lines in my "tsconfig.json":

```
  "include": [".astro/types.d.ts", "**/*"],
  "exclude": ["dist"],
```

So it was unable to import astro's types.

I figured it out to be my project becuase I went into other projects and they all worked.

@zvrkSam Thank you for your assistance!
